src/Mapping.elm

```diff
@@ -0,0 +1,396 @@
+
+
+--Exercise 15: Normalize Text
+--Task: Sanitize strings in a list by trimming whitespace
+--and converting text to lowercase.
+
+
+sanitizeString : String -> String
+sanitizeString string =
+    String.trim string
+        |> String.toLower
+
+
+sanitizeStrings : List String -> List String
+sanitizeStrings strings =
+    List.map sanitizeString strings
+
+
+--Exercise 16: Find Length
+--Task: Return the length of each string in a list.
+
+
+findLengths : List String -> List Int
+findLengths strings =
+    List.map String.length strings
+
+
+--Exercise 17: Append "day"
+--Task: Append "day" to the end of each string in a list
+--representing shortened or non-standard days of the week.
+
+
+appendDay : String -> String
+appendDay str =
+    str ++ "day"
+
+
+appendDayToList : List String -> List String
+appendDayToList strings =
+    List.map appendDay strings
+
+
+--Exercise 18: Convert Extended String to Boolean
+--Task: Convert each string in a list to its boolean
+--representation, considering extended variations such as
+--"true", "yes", "y", "1" for True and "false", "no", "n", "0" for False.
+--Case should not affect the conversion.
+
+
+stringToBool : String -> Bool
+stringToBool str =
+    case String.toLower str of
+        "true" ->
+            True
+
+        "yes" ->
+            True
+
+        "y" ->
+            True
+
+        "1" ->
+            True
+
+        _ ->
+            False
+
+convertStringsToBool : List String -> List Bool
+convertStringsToBool strings =
+    List.map stringToBool strings
+
+
+--Exercise 19: Calculate Future Ages
+--Task: Write a function that takes a specific year and a list of people
+--(each person having a name, year of birth, and gender), and returns a list
+--showing only the ages each person will be in that year. If a person was not born by that year,
+--their age should be reported as 0.
+
+
+
+type alias Person =
+    { name : String
+    , yearOfBirth : Int
+    , gender : String
+    }
+
+
+-- Function to calculate the age of a person in a specific year
+
+calculateAge : Int -> Person -> Int
+calculateAge currentYear person =
+    if currentYear >= person.yearOfBirth then
+        currentYear - person.yearOfBirth
+
+    else
+        0
+
+
+
+-- Function to calculate the ages of people in a list for a specific year
+
+
+calculateAges : Int -> List Person -> List Int
+calculateAges currentYear people =
+    List.map (calculateAge currentYear) people
+
+
+--Exercise 20: Calculate Future Ages (version 2)
+--Task: Write a function that takes a specific year and a list of people
+--(each person having a name, year of birth, and gender), and returns a list
+--showing tuples of each person's name and age they will be in that year.
+--If a person was not born by that year, their age should be reported as 0.
+--Type definition for a person
+-- Function to calculate the age of a person in a specific year
+
+
+ageCalculate : Int -> Person -> ( String, Int )
+ageCalculate currentYear person =
```

Comment:
Nice naming trick, ha ha.

However, since `'` is allowed in Elm variable names, you can call it `calculateAge'`.

That `'` is pronounced as __"prime"__, and is often used in Elm to show that some name points to a second version of something.